File: 프로그래머스/2/42842. 카펫/카펫.java
import java.util.*;
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];
        
        int total = brown + yellow; 
        
        for(int i = 1 ;i<=Math.sqrt(total); i++) {
            if(total%i == 0) {
                if(yellow == ((i-2)*(total/i-2))) {
                    answer[1] = i;
                    answer[0] = total/i;
                }
            }
        }
        return answer;
    }
}

File: 프로그래머스/2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    var answer = [];
    let sum = brown+yellow
    //의 약수 중간
    let width = Math.ceil(Math.sqrt(sum));
    
    
    while(true) {
       if(sum % width ===0){
           let height = sum/width;
           if((width-2) * (height -2) === yellow){
               return [width, height];
           }
       }
        width++;
    }
}